### DISCUSSION TITLE
Dependency error when upgrading from version 0.2.1 to later versions

### DISCUSSION THREAD

**Discussion by darxsky** [eclipse-edc/Connector#4096] [Q&A] [answered]
Hi,
I built my build.gradle for the creation of the .jar file. I am working on version 0.2.1 of the connector. I would like to upgrade it to later versions but I have problems with some dependencies, in particular the data-plane-framework dependency which no longer exists from version 0.2.1 onwards. I get this error. 

``` 
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':launchers:connector:shadowJar'.
> Could not resolve all dependencies for configuration ':launchers:connector:runtimeClasspath'.
> Could not resolve org.eclipse.edc:data-plane-framework:0.3.0.
Required by:
          project :launchers:connector
       > Could not resolve org.eclipse.edc:data-plane-framework:0.3.0.
          > Could not get resource 'https://maven.iais.fraunhofer.de/artifactory/eis-ids-public/org/eclipse/edc/data-plane-framework/0.3.0/data-plane-framework-0.3.0.pom'.
             > Could not GET 'https://maven.iais.fraunhofer.de/artifactory/eis-ids-public/org/eclipse/edc/data-plane-framework/0.3.0/data-plane-framework-0.3.0.pom'.
                > maven.iais.fraunhofer.de: Name or service not known
```

Since version 0.3.0, what should I replace that dependency with? If I remove the dependency, I get an error when starting the connector concerning the transfer. Is there a list of dependencies that need to be imported?

These are some of the dependencies I have in the build.gradle.kts: 

```
   // Control-plane to Data-plane
    api("${edcGroup}:transfer-data-plane:${edcVersion}")
    //api("${edcGroup}:transfer-pull-http-receiver:${edcVersion}")
    api("${edcGroup}:data-plane-selector-core:${edcVersion}")
    api("${edcGroup}:data-plane-selector-client:${edcVersion}")
    //api("${edcGroup}:data-plane-selector-api:${edcVersion}")

    // Data-plane
    api("${edcGroup}:data-plane-http:${edcVersion}")
    //api("${edcGroup}:data-plane-api:${edcVersion}")
    api("${edcGroup}:data-plane-framework:${edcVersion}")
    api("${edcGroup}:data-plane-core:${edcVersion}")
    api("${edcGroup}:data-plane-util:${edcVersion}")
```

Thanks for your support.

**Comment by ndr-brt** [comment]
Hi, there's no specific list, because the runtime can be composed in different ways depending from the needs.
What is the error message you receive? 
Eventually you could took a look at the [autodoc reference](https://eclipse-edc.github.io/Connector/autodoc/0.6.0/), that shows which extensions provide which service (please note that it is referring to version 0.6.0 because it is the first version of autodoc released, so it could not match 1:1 with 0.3.0)

**Reply by darxsky** [reply, marked as answer]
Hi,
thank you very much for the response. I took a look at the autodoc reference and solved the issue by inserting the dependency "control-plane-api-client".
